shared/protocol/src/main/java/io/pravega/shared/protocol/netty/AppendDecoder.java

```diff
@@ -95,28 +95,23 @@ public Request processCommand(WireCommand command) throws Exception {
                     ca.getExpectedOffset(), ca.getRequestId());
             break;
         case APPEND_BLOCK:
-            getSegment(((WireCommands.AppendBlock) command).getWriterId());
             currentBlock = (WireCommands.AppendBlock) command;
             result = null;
             break;
         case APPEND_BLOCK_END:
             WireCommands.AppendBlockEnd blockEnd = (WireCommands.AppendBlockEnd) command;
-            if (currentBlock == null) {
-                log.warn("Received AppendBlockEnd {} without AppendBlock", blockEnd);
-                throw new InvalidMessageException("AppendBlockEnd without AppendBlock.");
-            }
             UUID writerId = blockEnd.getWriterId();
-            if (!writerId.equals(currentBlock.getWriterId())) {
-                throw new InvalidMessageException("AppendBlockEnd for wrong connection.");
-            }
             segment = getSegment(writerId);
             if (blockEnd.getLastEventNumber() < segment.lastEventNumber) {
                 throw new InvalidMessageException("Last event number went backwards.");
             }
             int sizeOfWholeEventsInBlock = blockEnd.getSizeOfWholeEvents();
-            if (sizeOfWholeEventsInBlock > currentBlock.getData().readableBytes() || sizeOfWholeEventsInBlock < 0) {
+            if ((currentBlock != null) && (sizeOfWholeEventsInBlock > currentBlock.getData().readableBytes() || sizeOfWholeEventsInBlock < 0)) {
                 throw new InvalidMessageException("Invalid SizeOfWholeEvents in block");
```

Comment:
It would be useful in debugging these if you also included the values that lead you to this determination (`sizeOfWholeEventsInBlock`, etc.).

Reply:
This only for sizeOfWholeEventsInBlock value of append end block ; comparision happens only if its preceded by AppendBlock.